File: 2015/Day10/q.py
counter = 0
result = ""

data = "1113222113"

last = data[0]

for i in range(50):
    for char in data:
        if char == last:
            counter += 1
        else:
            result += str(counter) + last
            last = char
            counter = 1
    result += str(counter) + last

    print(f"Iteration {i+1}: " + result + '\n', "Length: " + str(len(result)) + '\n')

    data = result
    result = ""
    last = data[0]
    counter = 0

File: 2015/Day10/q.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string data = "1113222113", result = "";
    int counter = 0;
    char last = data[0];

    for (int i = 0; i < 50; i++) {
        for (int j = 0; j < data.length(); j++) {
            if (data[j] == last) {
                counter++;
            }
            else {
                result += to_string(counter) + last;
                last = data[j];
                counter = 1;
            }
        }
        result += to_string(counter) + last;

        cout << "Length: " << result.length() << "\n";

        data = result;
        result= "";
        last = data[0];
        counter = 0;
    }
}